symengine/derivative.cpp

```diff
@@ -72,32 +70,118 @@ class DiffImplementation
             return Derivative::create(self.rcp_from_this(), {x});
     }
 
-    static RCP<const Basic> diff(const Zeta &self, const RCP<const Symbol> &x)
+    // Needs create(vec_basic) method to be used.
+    template <typename T>
+    static RCP<const Basic> fdiff(const T &self, const RCP<const Symbol> &x)
     {
-        RCP<const Basic> m1 = self.get_s()->diff(x);
-        RCP<const Basic> diff
-            = mul(mul(mul(minus_one, self.get_s()),
-                      zeta(add(self.get_s(), one), self.get_a())),
-                  self.get_a()->diff(x));
-        if (eq(*m1, *zero)) {
-            return diff;
-        } else {
-            if (eq(*diff, *zero)) {
-                if (eq(*self.get_s(), *x)) {
+        RCP<const Basic> diff = zero;
+        RCP<const Basic> ret;
+        bool know_deriv;
+
+        vec_basic v = self.get_args();
+        vec_basic vdiff(v.size());
+
+        unsigned count = 0;
+        for (unsigned i = 0; i < v.size(); i++) {
+            vdiff[i] = v[i]->diff(x);
+            if (neq(*vdiff[i], *zero)) {
+                count++;
+            }
+        }
+
+        if (count == 0) {
+            return zero;
```

Comment:
Can we return `diff` here, just for aesthetics as we have already initialised it to `zero`? Otherwise it looks great. +1 to merge.